DeBlockWallet/NetworkManager.swift

```
@@ -0,0 +1,73 @@
+    
+    func fetchData() {
+        fetchEthereumPrice()
+        fetchGasPrice()
+    }
+    
+    private func fetchDataFromURL(_ urlString: String, completion: @escaping (Data) -> Void) {
+        guard let url = URL(string: urlString) else {
+            print("Invalid URL")
+            return
+        }
+        
+        let task = URLSession.shared.dataTask(with: url) { data, response, error in
+            guard let data = data, error == nil else {
+                print("Failed to fetch data: \(error?.localizedDescription ?? "Unknown error")")
+                return
+            }
+            completion(data)
+        }
+        
+        task.resume()
+    }
+    
+    func fetchEthereumPrice() {
+        fetchDataFromURL(ethPrice) { data in
+            do {
+                let decodedResponse = try JSONDecoder().decode(Crypto.self, from: data)
+                print("Ethereum Prices:")
+                print("USD: \(decodedResponse.ethereum.usd)")
+                print("EUR: \(decodedResponse.ethereum.eur)")
+                print("GBP: \(decodedResponse.ethereum.gbp)")
+            } catch {
+                print("Failed to decode JSON: \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    func fetchGasPrice() {
+        fetchDataFromURL(gasPrice) { data in
+            do {
+                let decodedResponse = try JSONDecoder().decode(ApiResponse.self, from: data)
+                guard let fastGasPrice = Double(decodedResponse.result.FastGasPrice) else {
+                    print("Error converting FastGasPrice to Double")
+                    return
+                }
+                let transactionFee = 21000.0 * fastGasPrice / 100000000.0
+                print("Approximate Ethereum transaction fee for a standard transfer with FastGasPrice: \(transactionFee) ETH")
+            } catch {
+                print("Failed to decode JSON:", error.localizedDescription)
```

Use closure, need to return an error and need to use completion handler to communicate the data between classes.